### PR TITLE
Qt project

### PR DESCRIPTION
**Pull request by cosunae**
# Technical Description

Some files, like the integration tests .cpp of gtclang are not included in Qt projects since they are not part of any cmake target. 
This creates functionality to add cpp files in a custom target (that does nothing) with the sole purpose that qt creator can add them to the project

This goes together with Meteoswiss-APN/gtclang#13

### CODE REVIEW

File: cmake/modules/DawnAddCustomDummyTarget.cmake [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+##   the purpose that qt creator can load those files into the project, which does not happen for those
+##   files that are not included in at least one target.
+##    
+##    NAME:STRING=<>             - Name of the dummy custom target
+##    DIRECTORIES:STRING=<>      - List of directories that contain the source files
+##
+function(dawn_add_custom_dummy_target)
+  # Parse arguments
+  set(options)
+  set(one_value_args GTCLANG NAME)
```
Comment (thfabian) [inline cmake/modules/DawnAddCustomDummyTarget.cmake]: This should be `set(one_value_args NAME)`

File: cmake/modules/DawnAddCustomDummyTarget.cmake [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  set(multi_value_args DIRECTORIES GRIDTOOLS_FLAGS)
```
Comment (thfabian) [inline cmake/modules/DawnAddCustomDummyTarget.cmake]: This should be `set(multi_value_args DIRECTORIES )`

File: cmake/modules/DawnAddCustomDummyTarget.cmake [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  cmake_parse_arguments(ARG "${options}" "${one_value_args}" "${multi_value_args}" ${ARGN})
+
+  set(all_files "")
+  foreach(dir ${ARG_DIRECTORIES})
+
+    file(GLOB test_sources "${dir}/*.cpp" )
+    file(GLOB test_headers "${dir}/*.hpp" )
```
Comment (thfabian) [inline cmake/modules/DawnAddCustomDummyTarget.cmake]: maybe also glob headers which end with `.h`?

File: cmake/modules/DawnAddCustomDummyTarget.cmake (hunk 2 above)
Comment (cosunae) [reply cmake/modules/DawnAddCustomDummyTarget.cmake]: done

File: cmake/modules/DawnAddCustomDummyTarget.cmake (hunk 1 above)
Comment (cosunae) [reply cmake/modules/DawnAddCustomDummyTarget.cmake]: done

File: cmake/modules/DawnAddCustomDummyTarget.cmake (hunk 3 above)
Comment (cosunae) [reply cmake/modules/DawnAddCustomDummyTarget.cmake]: done

**Comment by cosunae** [comment]
@thfabian I implemented your changes, are you fine? can I merge?

**Comment by thfabian** [comment]
lgtm

**Pull request merged**